File: app/src/main/java/com/moh/recipes/RecipeDatabase.java
package com.moh.recipes;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {Recipe.class}, version = 1)
public abstract class RecipeDatabase extends RoomDatabase {
    public abstract RecipeDao recipeDao();

    private static RecipeDatabase INSTANCE;

    public static RecipeDatabase getDbInstance(Context context) {

        if (INSTANCE == null) {
            // Initialize database
            INSTANCE = Room.databaseBuilder(context,
                    RecipeDatabase.class,
                    "recipes").build();
        }

        return INSTANCE;
    }
}

File: app/src/main/java/com/moh/recipes/RecipeDatabase.kt
package com.moh.recipes

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.Room.databaseBuilder
import androidx.room.RoomDatabase
import kotlinx.coroutines.CoroutineScope

@Database(entities = [Recipe::class], version = 1)
abstract class RecipeDatabase : RoomDatabase() {
    abstract fun recipeDao(): RecipeDao

    object DatabaseBuilder {
        private var INSTANCE: RecipeDatabase? = null
        fun getDbInstance(context: Context, scope: CoroutineScope): RecipeDatabase {
            return INSTANCE ?: synchronized(this) {
                INSTANCE = Room.databaseBuilder(context.applicationContext,
                        RecipeDatabase::class.java,
                        "recipes").build()
                INSTANCE!!
            }
        }
    }
}